commcare_connect/opportunity/forms.py

```diff
@@ -334,17 +335,48 @@ def __init__(self, *args, **kwargs):
             Row(Field("name")),
             Row(Field("description")),
             Row(Field("amount")),
-            Row(Field("deliver_units")),
+            Row(Field("required_deliver_units")),
+            Row(Field("optional_deliver_units")),
             Submit(name="submit", value="Submit"),
         )
 
         choices = [(deliver_unit.id, deliver_unit.name) for deliver_unit in deliver_units]
-        self.fields["deliver_units"] = forms.MultipleChoiceField(choices=choices, widget=forms.CheckboxSelectMultiple)
+        self.fields["required_deliver_units"] = forms.MultipleChoiceField(
+            choices=choices,
+            widget=forms.CheckboxSelectMultiple,
+            help_text="All of the selected Deliver Units are required for payment accrual.",
+        )
+        self.fields["optional_deliver_units"] = forms.MultipleChoiceField(
+            choices=choices,
+            widget=forms.CheckboxSelectMultiple,
+            help_text=(
+                "Any one of these Deliver Units combined with all the required "
+                "Deliver Units will accrue payment. Multiple Deliver Units can be selected."
+            ),
+            required=False,
+        )
         if PaymentUnit.objects.filter(pk=self.instance.pk).exists():
-            self.fields["deliver_units"].initial = [
-                deliver_unit.pk for deliver_unit in self.instance.deliver_units.all()
+            deliver_units = self.instance.deliver_units.all()
+            self.fields["required_deliver_units"].initial = [
+                deliver_unit.pk for deliver_unit in filter(lambda x: not x.optional, deliver_units)
+            ]
+            self.fields["optional_deliver_units"].initial = [
+                deliver_unit.pk for deliver_unit in filter(lambda x: x.optional, deliver_units)
             ]
 
+    def clean(self):
+        cleaned_data = super().clean()
+        if cleaned_data:
+            required_deliver_units = set(cleaned_data.get("required_deliver_units", []))
+            for deliver_unit in cleaned_data.get("optional_deliver_units", []):
```

Comment:
The code can be simplified by set operations 
```
         for unit in set(cleaned_data.get("required_deliver_units", [])) & set(cleaned_data.get("optional_deliver_units", [])):
                    deliver_unit_obj = DeliverUnit.objects.get(pk=unit)
                    self.add_error(
                        "optional_deliver_units",
                        error=f"{deliver_unit_obj.name} cannot be marked both Required and Optional",
                    )
```

Reply:
Fixed. https://github.com/dimagi/commcare-connect/pull/282/commits/f894d629ddc84d3ecafb888824b1f72e1b1dcf5f